resfit/fit_resonator/fit_S_data.py

```diff
@@ -788,18 +825,121 @@ class VNASweep:
+                amps = np.array(float(row[1]))
+                phases = np.array(float(row[2]))
+                line = snp.readline().strip()
+
+                while line:
+                    if row[0:1]=='!':
+                        line = snp.readline().strip()
+                        continue
+
+                    row = line.split()
+                    freqs = np.append(freqs,float(row[0]))
+                    amps = np.append(amps,float(row[1]))
+                    phases = np.append(phases,float(row[2]))
+                    line = snp.readline().strip()
+                phases = phases*np.pi/180
+                linear_amps = 10**(amps/20)
+
+            elif data_format == "ma":
+                freqs = np.array(float(row[0]))
+                linear_amps = np.array(float(row[1]))
+                phases = np.array(float(row[2]))
+                line = snp.readline().strip()
+
+                while line:
+                    if row[0:1]=='!':
+                        line = snp.readline().strip()
+                        continue
+
+                    row = line.split()
+                    freqs = np.append(freqs,float(row[0]))
+                    linear_amps = np.append(linear_amps,float(row[1]))
+                    phases = np.append(phases,float(row[2]))
+                    line = snp.readline().strip()
+                phases = phases*np.pi/180
+                amps = np.log10(linear_amps)*20
+
+            elif data_format == "ri":
+                freqs = np.array(float(row[0]))
+                real = np.array(float(row[1]))
+                imaginary = np.array(float(row[2]))
+                line = snp.readline().strip()
+
+                while line:
+                    if row[0:1]=='!':
+                        line = snp.readline().strip()
+                        continue
+
+                    row = line.split()
+                    freqs = np.append(freqs,float(row[0]))
+                    real = np.append(real,float(row[1]))
+                    imaginary = np.append(imaginary,float(row[2]))
+                    line = snp.readline().strip()
+                linear_amps = np.absolute(real + imaginary)
+                phases = np.angle(real + 1j*imaginary, deg=True)
+                amps = np.log10(linear_amps)*20
+
+            else:
+                print("Data type in file not supported. Please use DB, MA, or RI.")
+                quit()
+
+            if frequency_units == "hz":
+                freqs = freqs / 10**9
+            elif frequency_units == "khz":
+                freqs = freqs / 10**6
+            elif frequency_units == "mhz":
+                freqs = freqs / 10**3
+            elif frequency_units != "ghz":
+                print("Units for the frequency not found. Please include units for frequency in the header of the file.")
+            return cls(freqs=freqs,amps=amps,phases=phases,linear_amps=linear_amps)
+
+        else:
+            """Load data from other type of file."""
```

Comment:
You could put all of this into an `@classmethod` called `parse_csv(file)` and above into a `parcse_snp(file)`